src/db/model.rs

```diff
@@ -35,6 +42,12 @@ impl From<LockedChannel> for Document {
     }
 }
 
+impl From<SavedRoleEmbed> for Document {
+    fn from(saved_role_embed: SavedRoleEmbed) -> Self {
+        to_document(&saved_role_embed)
```

Comment:
To reduce duplicate code, create a trait and implement it for all the structs necessary.